apis/storage/v1alpha3/types.go

```diff
@@ -175,7 +175,7 @@ type LifecycleCondition struct {
 	//
 	// This condition is satisfied when an object is created before midnight of
 	// the specified date in UTC.
-	CreatedBefore metav1.Time `json:"createdBefore,omitempty"`
+	CreatedBefore *metav1.Time `json:"createdBefore,omitempty"`
```

Comment:
```suggestion
	// +optional
	CreatedBefore *metav1.Time `json:"createdBefore,omitempty"`
```

Also, it looks like we are not using our pointer syntax or `// +optional` directive anywhere in this type 🤔 I am guessing that the reason `createdBefore` is the only one that is failing is because it is using `metav1.Time`

Reply:
Done.
Yes, I also noticed that and agree that it is causing problems only with metav1.Time. 
It worths revisiting all fields and using pointers but I believe it should be out of scope for this bug fix. May be as part of https://github.com/crossplane/provider-gcp/issues/130

Reply:
Agreed 👍🏻